external/sparqlquery-service.js

```diff
@@ -0,0 +1,244 @@
+const axios = require('axios');
+const qs = require('qs');
+const constants = require('../modules/constants');
+const Engine = require('@comunica/query-sparql').QueryEngine;
+const N3 = require('n3');
+
+class SparqlqueryService {
+    constructor(config) {
+        this.config = config;
+    }
+
+    const;
+    filtertype = {
+        KEYWORD: 'keyword',
+        KEYWORDPREFIX: 'keywordPrefix',
+        TYPES: 'types',
+        ISSUERS: 'issuers'
+    };
+
+    async initialize(logger) {
+        this.logger = logger;
+        this.logger.info('Sparql Query module initialized successfully');
+        this.queryEngine = new Engine();
+    }
+
+    async insert(triples, rootHash) {
+        this.logger.info('dummy for ESLint');
+    }
+
+    async construct(query) {
+        const result = await this.executeQuery(query);
+        return result;
+    }
+
+    async ask(query) {
+        this.logger.info('dummy for ESLint');
+    }
+
+    async resolve(uri) {
+        const query = `PREFIX schema: <http://schema.org/>
+                        CONSTRUCT { ?s ?p ?o }
+                        WHERE {
+                          GRAPH <${constants.DID_PREFIX}:${uri}> {
+                            ?s ?p ?o
+                          }
+                        }`;
+        let nquads = await this.construct(query);
+
+        const writer = new N3.Writer();
+        writer.addQuads(nquads);
+        writer.end(async (error, result) => {
+            if (nquads.length) {
+                nquads = result;
+                nquads = nquads.split('\n');
+                nquads = nquads.filter((x) => x !== '');
+                nquads = await this.transformBlankNodes(nquads);
+            } else {
+                nquads = null;
+            }
+            return nquads;
+        });
+    }
+
+    async transformBlankNodes(nquads) {
```

Comment:
This transformation is only for Blazegraph so I think it should be left inside blazegraph service.

Reply:
Fixed in the latest commit. The resolve functionality has to be implemented by its subclass